### PR TITLE
Implement voice region changing for voice channels

### PR DESCRIPTION
**Pull request by Taarek**
## Summary

Adds the ability change the voice region of a voice channel, by passing either ``None`` (to set it to Automatic) or a string of the region name.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/channel.py [hunk 1]
```diff
@@ -609,7 +615,8 @@ def type(self):
 
     def _update(self, guild, data):
         self.guild = guild
-        self.name = data['name']
+        self.name = data.get('name')
+        self.rtc_region = data['rtc_region']
```
Comment (Rapptz) [inline discord/channel.py:619]:
```suggestion
        self.name = data['name']
        self.rtc_region = data.get('rtc_region')
        if self.rtc_region:
            self.rtc_region = try_enum(VoiceRegion, self.rtc_region)
```

This probably needs to be imported. I don't know why the `name` key went from regular `__getitem__` to `get`.

File: discord/guild.py [hunk 2]
```diff
@@ -844,6 +844,11 @@ def _create_channel(self, name, overwrites, channel_type, category=None, **optio
         except KeyError:
             pass
 
+        try:
+            options['rtc_region'] = options.pop('rtc_region')
+        except KeyError:
+            pass
```
Comment (Rapptz) [inline discord/guild.py:850]:
```suggestion
        try:
            rtc_region = options.pop('rtc_region')
        except KeyError:
            pass
        else:
            options['rtc_region'] = None if rtc_region is None else str(rtc_region)
```

File: discord/abc.py [hunk 3]
```diff
@@ -257,6 +257,11 @@ async def _edit(self, options, reason):
         except KeyError:
             pass
 
+        try:
+            options['rtc_region'] = options.pop('rtc_region')
+        except KeyError:
+            pass
```
Comment (Rapptz) [inline discord/abc.py:263]:
```suggestion
        try:
            rtc_region = options.pop('rtc_region')
        except KeyError:
            pass
        else:
            options['rtc_region'] = None if rtc_region is None else str(rtc_region)
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/guild.py [hunk 4]
```diff
@@ -840,7 +840,14 @@ def _create_channel(self, name, overwrites, channel_type, category=None, **optio
...
         try:
-            options['rate_limit_per_user'] = options.pop('slowmode_delay')
+            rtc_region = options.pop('rtc_region')
+        except KeyError:
+            pass
+        else:
+            options['rtc_region'] = None if rtc_region is None else str(rtc_region)
+
+        try:
+            options['rtc_region'] = options.pop('rtc_region')
```
Comment (Rapptz) [inline discord/guild.py:850]: There's no need to do this here.

File: discord/channel.py [hunk 5]
```diff
@@ -700,6 +709,10 @@ async def edit(self, *, reason=None, **options):
         overwrites: :class:`dict`
             A :class:`dict` of target (either a role or a member) to
             :class:`PermissionOverwrite` to apply to the channel.
+        rtc_region: Optional[:class:`VoiceRegion`]
+            The new region for the voice channel's voice communication.
+
+            .. versionadded:: 1.7
```
Comment (Rapptz) [inline discord/channel.py:715]:
```suggestion
            The new region for the voice channel's voice communication.
            A value of ``None`` indicates automatic voice region detection.

            .. versionadded:: 1.7
```

File: discord/channel.py [hunk 6]
```diff
@@ -575,10 +575,15 @@ class VoiceChannel(discord.abc.Connectable, discord.abc.GuildChannel, Hashable):
         The channel's preferred audio bitrate in bits per second.
     user_limit: :class:`int`
         The channel's limit for number of members that can be in a voice channel.
+    rtc_region: Optional[:class:`VoiceRegion`]
+        The new region for the voice channel's voice communication.
+
+        .. versionadded:: 1.7
```
Comment (Rapptz) [inline discord/channel.py:581]:
```suggestion
    rtc_region: Optional[:class:`VoiceRegion`]
        The region for the voice channel's voice communication.
        A value of ``None`` indicates automatic voice region detection.

        .. versionadded:: 1.7
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/guild.py (hunk 4 above)
Comment (Taarek) [reply discord/guild.py:850]: Sorry, thought I removed that.

File: discord/guild.py [hunk 7]
```diff
@@ -840,9 +840,11 @@ def _create_channel(self, name, overwrites, channel_type, category=None, **optio
             perms.append(payload)
 
         try:
-            options['rate_limit_per_user'] = options.pop('slowmode_delay')
+            rtc_region = options.pop('rtc_region')
         except KeyError:
             pass
+        else:
+            options['rtc_region'] = None if rtc_region is None else str(rtc_region)
```
Comment (Rapptz) [inline discord/guild.py:848]: Somehow the `slowmode_delay` handling got killed through this change.

**Review by Rapptz — CHANGES_REQUESTED** [review]

**Review by Rapptz — APPROVED** [review]
Thanks. LGTM. Sorry for the back and forth.

**Pull request merged**